internal/auth/auth_test.go

```diff
@@ -108,6 +111,101 @@ func TestRequiresTokenCheck(t *testing.T) {
 		})
 	}
 }
+
+func TestRequiresManufacturerTokenCheck(t *testing.T) {
+	t.Parallel()
+	mtrNFTAddrRaw := "0x1"
+	mtrNFTAddr := common.HexToAddress(mtrNFTAddrRaw)
+
+	validAutoPiAddr := common.BigToAddress(big.NewInt(123))
+	invalidAutoPiAddr := common.BigToAddress(big.NewInt(456))
+	id := NewMockIdentityService(gomock.NewController(t))
+	id.EXPECT().GetAftermarketDevice(gomock.Any(), &validAutoPiAddr, gomock.Any(), gomock.Any()).Return(
+		&identity.DeviceInfos{
+			ManufacturerTokenID: 137,
+		}, nil).AnyTimes()
+	id.EXPECT().GetAftermarketDevice(gomock.Any(), &invalidAutoPiAddr, gomock.Any(), gomock.Any()).Return(
+		nil, fmt.Errorf("")).AnyTimes()
+
+	tknValidator := &TokenValidator{
+		IdentitySvc: id,
+	}
+
+	testCases := []struct {
+		name               string
+		args               map[string]any
+		telmetryClaim      *TelemetryClaim
+		expectedError      error
+		tokenValidatorFunc func(ctx context.Context, _ any, next graphql.Resolver) (any, error)
+	}{
```

Comment:
I notice we don't test that the other methods work.

Reply:
what other methods are you referring to? the vehicle token check and privilege checks are split out into their own tests

Reply:
No like address and serial